src/dev/flang/fuir/GeneratingFUIR.java

```diff
@@ -0,0 +1,3431 @@
+/*
+
+This file is part of the Fuzion language implementation.
+
+The Fuzion language implementation is free software: you can redistribute it
+and/or modify it under the terms of the GNU General Public License as published
+by the Free Software Foundation, version 3 of the License.
+
+The Fuzion language implementation is distributed in the hope that it will be
+useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+License for more details.
+
+You should have received a copy of the GNU General Public License along with The
+Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+*/
+
+/*-----------------------------------------------------------------------
+ *
+ * Tokiwa Software GmbH, Germany
+ *
+ * Source of class FUIR
+ *
+ *---------------------------------------------------------------------*/
+
+package dev.flang.fuir;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import dev.flang.ast.AbstractAssign;
+import dev.flang.ast.AbstractBlock;
+import dev.flang.ast.AbstractCall;
+import dev.flang.ast.AbstractCurrent;
+import dev.flang.ast.AbstractFeature;
+import dev.flang.ast.AbstractMatch;
+import dev.flang.ast.AbstractType;
+import dev.flang.ast.Box;
+import dev.flang.ast.Constant;
+import dev.flang.ast.Context;
+import dev.flang.ast.Env;
+import dev.flang.ast.Expr;
+import dev.flang.ast.FeatureName;
+import dev.flang.ast.InlineArray;
+import dev.flang.ast.NumLiteral;
+import dev.flang.ast.ResolvedNormalType;
+import dev.flang.ast.Tag;
+import dev.flang.ast.Types;
```

Comment:
I think the import Types should eventually be removed?

Reply:
All imports from `dev.flang.ast` should be replaced. by a new package that contains code common to `ast` and `fe`, so there should eventually be no more dependencies on `dev.flang.ast